sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -195,12 +197,29 @@ void gnrc_rpl_send_DIS(gnrc_rpl_dodag_t *dodag, ipv6_addr_t *destination)
     _gnrc_rpl_send(pkt, NULL, destination, (dodag ? &dodag->dodag_id : NULL));
 }
 
+static bool _gnrc_rpl_check_DIS_validity(gnrc_rpl_dis_t *dis, uint16_t len)
+{
+    uint16_t expected_len = sizeof(*dis) + sizeof(icmpv6_hdr_t);
+
+    if (expected_len <= len) {
+        return true;
+    }
+
+    DEBUG("RPL: wrong DIS len: %d, expected: %d\n", len, expected_len);
+
+    return false;
+}
+
 void gnrc_rpl_recv_DIS(gnrc_rpl_dis_t *dis, ipv6_addr_t *src, ipv6_addr_t *dst, uint16_t len)
 {
     /* TODO handle Solicited Information Option */
     (void) dis;
     (void) len;
 
+    if (!_gnrc_rpl_check_DIS_validity(dis, len)) {
```

Comment:
If this is the only use case for `_gnrc_rpl_check_DIS_validity()`, I would simply put the comparison here instead of introducing a function.